### ISSUE TITLE
Starting from 9.0.12 TitleBar Missing From Modal Pages

### ISSUE THREAD

**Issue by olographio** [dotnet/maui#26690]
### Description

Starting from 9.0.12 the title bar appears to be missing from modal pages when using a NavigationPage as Window.Page.

Basically in app.xaml.cs, I instantiate a window with a custom titlebar. This window's main page is a NavigationPage. The main page of the Navigation page is a TestPage that I created with a push and pop button and a modal page counter. The TestPage can thus open more instance of iteself. It uses NavigationPage.Navigation.PushModalAsync().

When using version ealier to 9.0.12, the custom titlebar is always present. From version 9.0.12, the titlebar disappears when opening modal pages. Only the Window.Title property still works. The titlebar is however gone altogether.

### Steps to Reproduce

Please look at the repro repository source. Simply start one of the project. Use the push and pop button. You'll see that for the TitleBarRepro.Issue project, the titlebar disappear when you push/open a page.

### Link to public reproduction project repository

https://bitbucket.org/ba81f149/titlebarrepro/src/main/

### Version with bug

9.0.12 SR1.2

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

9.0.10 SR1

### Affected platforms

Windows

### Affected platform versions

net9.0-windows10.0.19041.0

### Did you find any workaround?

I found no workaround, however, when using the "XAML Hot Reload Toolbar" one can select an element. Please select the window titlebar. In the Live Visual Tree, you'll find several instances of AppTitleBarContainer... The first one's visibility is set to "Collapsed". Set it to "Visible" and the title bar will show properly.

 ![Image](https://github.com/user-attachments/assets/6dee0be2-a522-4b4c-be97-acc35b852a24)

### Relevant log output

```shell

```

**Comment by kevinxufei** [comment]
I can repro this issue at Windows platform on the latest 17.13 preview 2(9.0.12 & 9.0.21), but it works fine on 9.0.10.

**Comment by olographio** [comment]
> I can repro this issue at Windows platform on the latest 17.13 preview 2(9.0.12 & 9.0.21), but it works fine on 9.0.10.

Precisely. I tested with 9.0.21 too and I can confirm the issue was there. FYI the repo contains twice the same project, one uses 9.0.10 (works fine) and 9.0.12 (issue).

**Comment by bcaceiro** [comment]
It is also missing when doing Shell.GoToAsync() and the page has PResentationMode="Modal" /"ModalNotAnimated"
